discord/ext/tasks/__init__.py

```diff
@@ -166,7 +183,27 @@ async def _loop(self, *args: Any, **kwargs: Any) -> None:
                     self._last_iteration = self._next_iteration
                     self._next_iteration = self._get_next_sleep_time()
                 try:
-                    await self.coro(*args, **kwargs)
+                    token = _current_loop_ctx.set(self._current_loop)
+                    if not self.overlap:
+                        await self.coro(*args, **kwargs)
+                    else:
+
+                        async def run_with_semaphore():
```

Comment:
maybe could find a way to avoid redefining the function every time idk with a separate function or smth